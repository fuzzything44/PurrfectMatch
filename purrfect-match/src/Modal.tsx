import React from 'react';

interface ModalProps {
    display: boolean;
    onClose: () => void
};

interface ModalState {

};

export class Modal extends React.Component<ModalProps, ModalState> {

    public render() {
        if (!this.props.display) {
            return null;
        }
        return (<div className="modal-overlay" onClick={() => this.props.onClose()}>
            <div className="modal" onClick={(e) => e.stopPropagation() }>
                <button className="modal-close" onClick={() => this.props.onClose()} >X</button>
                {this.props.children}
            </div>
        </div>);
    }

}


