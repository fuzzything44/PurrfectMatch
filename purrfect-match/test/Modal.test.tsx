import React from 'react';
import ReactDOM from 'react-dom';
import { configure, shallow, ShallowWrapper } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Modal } from 'Modal';

configure({ adapter: new Adapter() });

let wrapper: ShallowWrapper<any, any, any>;
beforeEach(() => {
    wrapper = shallow(<Modal display={true} onClose={jest.fn()}><div>mockContent</div></Modal>);
});

test("Renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
    wrapper.setProps({ display: false });
    expect(wrapper).toMatchSnapshot();
});

test("Closes when clicked", () => {
    let stopPropagation = jest.fn();
    wrapper.find(".modal").simulate("click", { stopPropagation: stopPropagation });
    expect(stopPropagation).toHaveBeenCalled();

    wrapper.find(".modal-overlay").simulate("click", {});
    expect(wrapper.instance().props.onClose).toHaveBeenCalled();

    wrapper.find(".modal-close").simulate("click", {});
    expect(wrapper.instance().props.onClose).toHaveBeenCalledTimes(2);
});