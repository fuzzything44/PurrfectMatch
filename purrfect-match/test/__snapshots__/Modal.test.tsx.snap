// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<div
  className="modal-overlay"
  onClick={[Function]}
>
  <div
    className="modal"
    onClick={[Function]}
  >
    <button
      className="modal-close"
      onClick={[Function]}
    >
      X
    </button>
    <div>
      mockContent
    </div>
  </div>
</div>
`;

exports[`Renders correctly 2`] = `""`;
